core/computed/entity-classification.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license Copyright 2022 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {makeComputedArtifact} from './computed-artifact.js';
+import {NetworkRecords} from './network-records.js';
+import {Util} from '../util.cjs';
+import UrlUtils from '../lib/url-utils.js';
+import thirdPartyWeb from '../lib/third-party-web.js';
+
+/** @typedef {Record<string, LH.Artifacts.RecognizableEntity>} EntityCache */
+
+class EntityClassification {
+  /**
+   * @param {EntityCache} entityCache
+   * @param {string} url
+   * @return Entity | undefined
+   */
+  static makeUpAnEntity(entityCache, url) {
+    if (!UrlUtils.isValid(url)) return undefined;
+    const rootDomain = Util.getRootDomain(url);
+    if (!rootDomain) return undefined;
+    if (rootDomain in entityCache) return entityCache[rootDomain];
+
+    return /** @type LH.Artifacts.RecognizableEntity */ (entityCache[rootDomain] = {
+      name: rootDomain,
+      company: rootDomain,
+      category: '',
+      categories: [],
+      domains: [rootDomain],
+      averageExecutionTime: 0,
+      totalExecutionTime: 0,
+      totalOccurrences: 0,
+      isUnrecognized: true,
+    });
+  }
+
+  /**
+   *
+   * @param {EntityCache} entityCache
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @return {{
```

Comment:
Not sure how @brendankenny would feel about this and I'd defer to him, but I'd consider dropping the `@return` here (and it's awkward formatting) and rely on the inferred return type. Or, can make a typedef jsdoc at top of file.

Reply:
I think I can rewrite to eliminate that function and move to `compute_`. Will update the PR.